crates/ln-dlc-node/src/fee_rate_estimates.rs

```diff
@@ -0,0 +1,100 @@
+use anyhow::Result;
+use autometrics::autometrics;
+use bdk::FeeRate;
+use lightning::chain::chaininterface::ConfirmationTarget;
+use lightning::chain::chaininterface::FeeEstimator;
+use lightning::chain::chaininterface::FEERATE_FLOOR_SATS_PER_KW;
+use std::collections::HashMap;
+use tokio::sync::RwLock;
+
+pub struct FeeRateEstimates {
+    client: esplora_client::AsyncClient,
+    fee_rate_cache: RwLock<HashMap<ConfirmationTarget, FeeRate>>,
+    fallbacks: RwLock<FeeRateFallbacks>,
+    runtime_handle: tokio::runtime::Handle,
+}
+
+#[derive(Debug)]
+pub struct FeeRateFallbacks {
+    pub normal_priority: u32,
+    pub high_priority: u32,
+}
+
+impl Default for FeeRateFallbacks {
+    fn default() -> Self {
+        Self {
+            normal_priority: 2000,
+            high_priority: 5000,
+        }
+    }
+}
+
+impl FeeRateEstimates {
+    pub fn new(esplora_url: String, runtime_handle: tokio::runtime::Handle) -> Self {
+        let client = esplora_client::AsyncClient::from_client(esplora_url, reqwest::Client::new());
+
+        let fee_rate_cache = RwLock::new(HashMap::default());
+        let fallbacks = RwLock::new(FeeRateFallbacks::default());
+
+        Self {
+            client,
+            fee_rate_cache,
+            fallbacks,
+            runtime_handle,
+        }
+    }
+
+    pub(crate) fn get(&self, target: ConfirmationTarget) -> FeeRate {
+        let (cache, fallbacks) = self.runtime_handle.block_on(async {
+            (
+                self.fee_rate_cache.read().await,
+                self.fallbacks.read().await,
+            )
+        });
+
+        cache.get(&target).copied().unwrap_or_else(|| {
+            let fee_rate = match target {
+                ConfirmationTarget::Background => FEERATE_FLOOR_SATS_PER_KW,
+                ConfirmationTarget::Normal => fallbacks.normal_priority,
+                ConfirmationTarget::HighPriority => fallbacks.high_priority,
+            };
+            FeeRate::from_sat_per_kwu(fee_rate as f32)
+        })
+    }
+
+    #[autometrics]
+    pub(crate) async fn update(&self) -> Result<()> {
+        let mut locked_fee_rate_cache = self.fee_rate_cache.write().await;
+
+        let estimates = self.client.get_fee_estimates().await?;
+
+        for (target, n_blocks) in [
+            (ConfirmationTarget::Background, 12),
+            (ConfirmationTarget::Normal, 6),
+            (ConfirmationTarget::HighPriority, 3),
+        ] {
+            let fee_rate = esplora_client::convert_fee_rate(n_blocks, estimates.clone())?;
+            let fee_rate = FeeRate::from_sat_per_vb(fee_rate);
+
+            locked_fee_rate_cache.insert(target, fee_rate);
+            tracing::trace!(
+                n_blocks_confirmation = %n_blocks,
+                sats_per_kwu = %fee_rate.fee_wu(1000),
+                "Updated fee rate estimate",
+            );
+        }
+        Ok(())
+    }
+
+    pub async fn update_fallbacks(&self, fallbacks: FeeRateFallbacks) {
```

Comment:
I believe you forgot to update it when new settings come from the coordinator - we lost the ability to influence the fallbacks, even at startup :(

Reply:
Yep, I missed that. I've rectified this now.